avm/utilities/pipelines/sharedScripts/Get-AzRecommendedRegions.ps1

```diff
@@ -0,0 +1,93 @@
+<#
+.SYNOPSIS
+Get the recommended regions for deployment
+
+.DESCRIPTION
+This script is used to set the deployment region for the mmodule deployment.
+
+.PARAMETER excludedRegions
+Optional. The regions to exclude from the recommended regions list.
+
+.PARAMETER usePairedRegionsOnly
+Optional. If set, only paired regions will be returned.
+
+.PARAMETER useKnownRegionsOnly
+Optional. If set, only known regions will be returned.
+
+.PARAMETER knownRegionList
+Optional. The list of known regions to use when useKnownRegionsOnly is set. ARM Deployment are limited to 10 active deployment regions per subscription.
+
+.EXAMPLE
+Get-AzRecommendedRegions -excludedRegions @("West Europe", "Brazil South", "West US 2", "East US 2", "South Central US") -usePairedRegionsOnly -useKnownRegionsOnly
+
+Get the recommended regions using known regions excluding the regions specified in the excludedRegions parameter and only paired regions.
+
+
+.EXAMPLE
+Get-AzRecommendedRegions -excludedRegions @("West Europe", "Brazil South", "West US 2", "East US 2", "South Central US") -usePairedRegionsOnly
+
+Get the recommended regions excluding the regions specified in the excludedRegions parameter and only paired regions.
+
+.EXAMPLE
+Get-AzRecommendedRegions -excludedRegions @("West Europe", "Brazil South", "West US 2", "East US 2", "South Central US")
+
+Get the recommended regions excluding the regions specified in the excludedRegions parameter.
+
+.EXAMPLE
+Get-AzRecommendedRegions -usePairedRegionsOnly
+
+Get the recommended paired regions.
+#>
+
+function Get-AzRecommendedRegions {
+  param (
+    [Parameter(Mandatory = $false)]
+    [object]$excludedRegions = @(),
+    [Parameter(Mandatory = $false)]
+    [switch]$usePairedRegionsOnly,
+    [Parameter(Mandatory = $false)]
+    [switch]$useKnownRegionsOnly,
+    [Parameter(Mandatory = $false)]
+    [object]$knownRegionList = @()
+  )
+
+  $allRegions = Get-AzLocation -ExtendedLocation $true
```

Comment:
This function got me thinking a bit - we do have a list of locations we want - and a list of locations we don't want. Why even both calling the `Get-AzLocation` function and not just apply a random selector on the array of known regions?

Reply:
Using the AzLocation its polling the latest information from the API so if a paired region ever goes down or we no longer recommend a region (returned value from API) then we want to exclude them from the list

Reply:
I see, I see. Thanks for elaborating